assets/src/design-system/components/contextMenu/index.js

```diff
@@ -0,0 +1,162 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useMemo } from 'react';
+import styled, { css } from 'styled-components';
+import { v4 as uuidv4 } from 'uuid';
+/**
+ * Internal dependencies
+ */
+import { BUTTON_TRANSITION_TIMING } from '../button/constants';
+import { MenuItem, MenuItemProps } from './menu-item';
+
+const Popover = styled.div(
+  ({ isOpen, theme }) => css`
+    position: absolute;
+    display: none;
+    margin: 5px 0 0;
+    background-color: ${theme.colors.bg.primary};
+    pointer-events: none;
+    border-radius: ${theme.borders.radius.small};
+    border: 1px solid ${theme.colors.border.disable};
+    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+
+    ${isOpen &&
+    css`
+      display: block;
+      z-index: 10;
+      opacity: 1;
+      pointer-events: auto;
+    `};
+  `
+);
+
+const MenuContainer = styled.ul(
+  ({ theme }) => css`
+    background-color: ${theme.colors.bg.primary};
+    border-radius: ${theme.borders.radius.small};
+    margin: 0;
+    min-width: 210px;
+    padding: 5px 0;
+    pointer-events: auto;
+    list-style: none;
+
+    a {
+      background-color: transparent;
+      text-decoration: none;
+    }
+
+    li {
+      a,
+      button,
+      div {
+        display: flex;
+        justify-content: space-between;
+        align-items: center;
+        padding: 10px 18px;
+        border: 0;
+        transition: background-color ${BUTTON_TRANSITION_TIMING};
+      }
+
+      span {
+        transition: color ${BUTTON_TRANSITION_TIMING};
+      }
+
+      a {
+        :active span,
+        :hover span {
+          color: ${theme.colors.fg.primary};
+        }
+      }
+
+      button {
+        width: 100%;
+        border-radius: 0;
+        background-color: transparent;
+
+        :disabled {
+          background-color: transparent;
+
+          span {
+            color: ${theme.colors.bg.tertiary};
+          }
+        }
+
+        :active span,
+        :hover span {
+          color: ${theme.colors.fg.primary};
+        }
+      }
+
+      &.separatorTop {
+        border-top: 1px solid ${theme.colors.bg.tertiary};
+      }
+
+      &.separatorBottom {
+        border-bottom: 1px solid ${theme.colors.bg.tertiary};
+      }
+
+      :hover a,
+      button:hover:not(:disabled) {
+        background-color: ${theme.colors.interactiveBg.secondaryHover};
+      }
+
+      :active a,
+      button:active:not(:disabled) {
+        background-color: ${theme.colors.interactiveBg.secondaryPress};
+      }
+    }
+  `
+);
+
+const ContextMenu = ({ isOpen, isLTR, items, ...props }) => {
+  const ids = useMemo(() => items.map(() => uuidv4()), [items]);
+
+  return (
+    <Popover isOpen={isOpen} {...props}>
+      <MenuContainer>
```

Comment:
I _feel like_ this could benefit from an aria label,  samesies with the list items since the spans inside aren't getting picked up in the ol' accessibility tree. 
<img width="767" alt="Screen Shot 2021-02-11 at 1 47 44 PM" src="https://user-images.githubusercontent.com/10720454/107696877-bba11d80-6c6f-11eb-9a56-3234b239fbc9.png">

Reply:
Chatted with Brittany about this but posting here for visibility:

When adding a label to the list item the label gets repeated twice since screenreaders read the list item and then the button/link inside of it. We can't use `aria-hidden` on the inner content since they are focusable. 

Looked around online to see what the standard is here. Didn't find much, but it looks like these people don't label the list items either: http://web-accessibility.carnegiemuseums.org/code/navigation/

For now we'll leave it off